ee/clickhouse/models/test/test_cohort.py

```diff
@@ -852,7 +817,40 @@ def test_query_with_multiple_new_style_cohorts(self):
 
         cohort1.calculate_people_ch(pending_version=0)
 
-        result = sync_execute(
-            "SELECT person_id FROM cohortpeople where cohort_id = %(cohort_id)s", {"cohort_id": cohort1.pk}
-        )
+        result = self._get_cohortpeople(cohort1)
         self.assertCountEqual([p1.uuid, p3.uuid], [r[0] for r in result])
+
+    def test_update_cohort(self):
+        p1 = Person.objects.create(team_id=self.team.pk, distinct_ids=["1"], properties={"$some_prop": "something"},)
+
+        p2 = Person.objects.create(team_id=self.team.pk, distinct_ids=["2"], properties={"$another_prop": "something"},)
+
+        p3 = Person.objects.create(team_id=self.team.pk, distinct_ids=["3"], properties={"$another_prop": "something"},)
+
+        cohort1 = Cohort.objects.create(
+            team=self.team,
+            groups=[{"properties": [{"key": "$some_prop", "value": "something", "type": "person"},]}],
+            name="cohort1",
+        )
+
+        cohort1.calculate_people_ch(pending_version=0)
+
+        # Should only have p1 in this cohort
+        results = self._get_cohortpeople(cohort1)
+        self.assertEqual(len(results), 1)
+
+        cohort1.groups = [{"properties": [{"key": "$another_prop", "value": "something", "type": "person"},]}]
+        cohort1.save()
+        cohort1.calculate_people_ch(pending_version=1)
+
+        # Should only have p2, p3 in this cohort
+        results = self._get_cohortpeople(cohort1)
+        self.assertEqual(len(results), 2)
+
+        cohort1.groups = [{"properties": [{"key": "$some_prop", "value": "something", "type": "person"},]}]
+        cohort1.save()
+        cohort1.calculate_people_ch(pending_version=2)
+
+        # Should only have p1 again in this cohort
+        results = self._get_cohortpeople(cohort1)
+        self.assertEqual(len(results), 1)
```

Comment:
It'd be great to add a test that breaks if we add a `where version = X` to the select statement in the future. I think this is just requires:
* version = null on the cohortpeople table
* version of the cohort > 0